pkg/healthz/healthz_test.go

```diff
@@ -339,6 +335,11 @@ func (o *fakeResourceLock) Get(ctx context.Context) (*resourcelock.LeaderElectio
 	if err != nil {
 		return nil, nil, err
 	}
+	if o.record.HolderIdentity == remoteIdentity {
+		// Always return unique leader election record bytes,
+		// to simulate an always updated remote leader election record.
+		lerByte = []byte(uuid.NewUUID())
```

Comment:
```go
	if o.record == nil {
		err := errors.NewNotFound(schema.ParseGroupResource("configmap"), "foo")
		return nil, nil, err
	}
	// If the current leader is not the local node, return a unique leader election record.
	// This simulates the case where the current leader is active.
	if o.record.HolderIdentity != localIdentity {
		o.record.LeaderTransitions++
	}
	lerByte, err := json.Marshal(*o.record)
	if err != nil {
		return nil, nil, err
	}
```

@wallrj could we do something like this instead? It makes it less abstract for me & easier to understand.

Reply:
I considered that, but had assumed that it would trigger the `OnNewLeader` callback each time, which I wanted to avoid doing.

UUID is reasonable here, and aligns with the comments in the leader election source code.

// > LeaderElectionRecord is the record that is stored in the leader election annotation.
// > This information should be used for observational purposes only and could be replaced
// > with a random string (e.g. UUID) with only slight modification of this code.
// > -- https://github.com/kubernetes/kubernetes/blob/7e25f1232a9f89875641431ae011c916f0376c57/staging/src/k8s.io/client-go/tools/leaderelection/re

But looking again at the code, I see that that callback is triggered only by the change of the leader name:

https://github.com/kubernetes/kubernetes/blob/7e25f1232a9f89875641431ae011c916f0376c57/staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go#L372-L380

So yes, that would probably also work.

Reply:
Great, I don't think I care much about which becomes the final solution. I do think we need to explain better in the comments why we do what we do.